### PR TITLE
[CI] Clean workspace on self-hosted runners on start

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
We need to clean the workspace on self-hosted runners before executing build, because it may contain old builds, which poison the current build.

### CODE REVIEW

File: .github/workflows/pr_push.yml [hunk 1]
```diff
@@ -15,87 +15,5 @@ permissions:
   contents: read
 
 jobs:
-  CodeStyle:
```
Comment (lplewa) [inline .github/workflows/pr_push.yml:18]: Do you intend this?

Comment (lukaszstolarczuk) [reply .github/workflows/pr_push.yml:18]: haha, rather so😄

I'm testing the GPU workflow only

**Comment by lukaszstolarczuk** [comment]
cleaning workspace does not solve the issue, it still occurs from time to time, with error:

```
CUSTOMBUILD : CMake error : Cannot restore timestamp "C:/Users/Administrator/repo/actions-runner/_work/unified-memory-framework/unified-memory-framework/build/test/CMakeFiles/generate.stamp": Access is denied. [C:\Users\Administrator\repo\actions-runner\_work\unified-memory-framework\unified-memory-framework\build\test\umf_test-jemalloc_pool.vcxproj]
```

we'd have to investigate this further to avoid these random fails

**Comment by KFilipek** [comment]
As we talked few weeks ago, it should be done only via: https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/running-scripts-before-or-after-a-job 
Hooks will always clean the working directory before/after.
Additionally, this solution is done purely in infrastructure, nothing to do with PR.

**Comment by lukaszstolarczuk** [comment]
This change does not fix the issue as we expected. Closing it for now.

**Pull request closed without merging**